datafusion/core/src/physical_plan/aggregates/hash.rs

```diff
@@ -498,14 +502,26 @@ fn create_batch_from_map(
         }
     }
 
+    // make group states mutable
+    let (mut group_by_values_vec, mut accumulator_set_vec): (Vec<_>, Vec<_>) =
+        accumulators
+            .group_states
+            .into_iter()
+            .map(|group_state| {
+                (
+                    VecDeque::from(group_state.group_by_values.to_vec()),
```

Comment:
🤔  maybe we could use a VecDeque always and could avoid this copy too 🤔

Reply:
It's not a copy. Due to move semantics, Rust Std should just reuse the pointer to the allocated backing array.

Reply:
Relevant code:

https://github.com/rust-lang/rust/blob/a803f313fdf8f6eb2d674d7dfb3694a2b437ee1e/library/alloc/src/collections/vec_deque/mod.rs#L2833-L2880